browser/extensions/api/brave_shields_api.cc

```diff
@@ -54,60 +54,90 @@ BraveShieldsUrlCosmeticResourcesFunction::Run() {
   std::unique_ptr<brave_shields::UrlCosmeticResources::Params> params(
       brave_shields::UrlCosmeticResources::Params::Create(*args_));
   EXTENSION_FUNCTION_VALIDATE(params.get());
+  g_brave_browser_process->ad_block_service()->GetTaskRunner()
```

Comment:
@antonok-edm This is not super illustrative example of async code - here it is implicitly asserted that all service run on the same task runner exposed by any of them. It would be better for this services to expose async api accepting callbacks for passing resulted values, dealing with task runners inside them.

Reply:
this doesn't seem like the right place to do this, it's going to return the value on the wrong thread now

Reply:
@bridiver could you please elaborate